test.js

```diff
@@ -0,0 +1,124 @@
+const app = require('express')()
+const bodyParser = require('body-parser')
+const multer = require('multer') // v1.0.5
+const upload = multer() // for parsing multipart/form-data
+
+app.use(bodyParser.json()) // for parsing application/json
+app.use(bodyParser.urlencoded({ extended: true })) // for parsing application/x-www-form-urlencoded
+
+app.post('/profile', upload.array(), function (req, res, next) {
+  console.log(req.body)
+  res.json(req.body)
+})
+
+const port = 3001;
+
+app.get('/', (req, res) => res.send('Hello World!'));
+
+
+class Treenode {
+
+  constructor(label) {
+    this.id = (++Treenode.id_gen).toString()
+    // Treenode.lineage[this.id]=[label]
+    this.label =  label
+    this.parent_id = null
+    // this.parent_label = null
+    this.children = []
```

Comment:
nodes that define relationships twoways (parentId + children) are convenient when querying, but inconvenient when updating with a large responsibility and importance to keep both fields to be accurate